pkg/registries/ecr/ecr.go

```diff
@@ -84,10 +89,19 @@ func sanitizeConfiguration(ecr *storage.ECRConfig) error {
 	return errorList.ToError()
 }
 
+func (e *ecr) setLastRefreshAttempt() {
+	e.lastRefreshAttempt = time.Now()
+}
+
 func (e *ecr) refreshDockerClient() error {
-	if e.expiryTime.After(time.Now()) {
+	e.lock.Lock()
+	defer e.lock.Unlock()
+
+	if e.expiryTime.After(time.Now()) || time.Since(e.lastRefreshAttempt) < refreshAttemptInterval {
```

Comment:
@jvdm Do you know what the normal expiry time is for these tokens? I would assume over 30s but just want to make sure

Reply:
They have a lifespan of 12hs:

> GetAuthorizationToken
[PDF](https://docs.aws.amazon.com/pdfs/AmazonECR/latest/APIReference/ecr-api.pdf#API_GetAuthorizationToken)
Retrieves an authorization token. An authorization token represents your IAM authentication credentials and can be used to access any Amazon ECR registry that your IAM principal has access to. The authorization token is valid for 12 hours.

Reply:
Gotcha. If you could take a quick look and make sure this PR seems reasonable then I'll ship it :)